src/dbnode/client/host_queue.go

```diff
@@ -879,6 +885,20 @@ func (q *queue) asyncFetchTagged(op *fetchTaggedOp) {
 		ctx, _ := thrift.NewContext(q.opts.FetchRequestTimeout())
 		result, err := client.FetchTagged(ctx, &op.request)
 		if err != nil {
+			if se, ok := err.(tchannel.SystemError); ok {
+				if se.Code() == tchannel.ErrCodeUnexpected &&
+					strings.Contains(se.Message(), rpc.ErrorType_BAD_REQUEST.String()) {
+
+					newErr, parseErr := tterrors.ParseErrorFromString(se.Message())
+					if parseErr == nil {
+						err = newErr
+					} else {
+						q.logger.Debug("failed to parse application error from error string. " +
+							"ignoring and using provided system error")
```

Comment:
Probably worth logging the error anyway, especially since it's a `Debug`?

```suggestion
						q.logger.Debug("failed to parse application error from error string. " +
							"ignoring and using provided system error",
							zap.String("application error", se.Message()))
```